src/metabase/models/dashboard.clj

```diff
@@ -131,6 +141,23 @@
   [dashboard]
   (update-dashboard-subscription-pulses! dashboard))
 
+(defn- card-ids-by-param-id
+  "Returns a map from parameter IDs to card IDs for the given dashboard (for parameters whose filter values come from a
+  card via a ParameterCard)."
+  [dashboard-id]
+  (->> (db/select ParameterCard :parameterized_object_id dashboard-id :parameterized_object_type "dashboard")
+       (group-by :parameter_id)
+       (m/map-vals (comp :card_id first))))
+
+(defn- populate-card-id-for-parameters
```

Comment:
I'm not entirely sure why Tim did this, we've been discussing this [here](https://github.com/metabase/metabase/pull/27008#discussion_r1049735046).

My understanding is when selecting a dashboard, we already have `card_id` in `dashboard.paramteres[0].source_options`, and the values in the list should be authoritative because `dashboard.paramters` is the source of truth.

I'm not sure why we need to re-populate it here.